### PR TITLE
Add Dynakube condition for Extensions TLS reconciler

### PR DESCRIPTION
**Pull request by albertogdd**
## Description

https://dt-rnd.atlassian.net/browse/DAQ-1717

Add new condition `ExtensionsTLSSecret` when Extensions TLS reconciler generates a self-signed TLS secret

## How can this be tested?

- Deploy Dynakube with extensions enabled and without `tlsRefName`, so the Operator generates the secret and adds the condition.
```yaml
spec:
  apiUrl: https://dib18963.dev.dynatracelabs.com/api
  extensions:
    enabled: true
  templates:
    extensionExecutionController:
      imageRef:
        repository: extk8sregistry.azurecr.io/eec/dynatrace-eec
        tag: 1.301.0.20240826-070516
  customPullSecret: azurecr
  tokens: dynakube
```

### CODE REVIEW

File: pkg/controllers/dynakube/extension/tls/reconciler_test.go [hunk 1]
```diff
@@ -64,10 +66,16 @@ func TestReconcile(t *testing.T) {
 
 		require.NoError(t, err)
 		assert.NotEmpty(t, secret)
+		assert.NotEmpty(t, dk.Conditions())
```
Comment (albertogdd) [inline pkg/controllers/dynakube/extension/tls/reconciler_test.go:69]:
```suggestion
		require.NotEmpty(t, dk.Conditions())
```

File: pkg/controllers/dynakube/extension/tls/conditions.go [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package tls
+
+const extensionsTLSSecretConditionType string = "ExtensionsTLSSecret"
```
Comment (0sewa0) [inline pkg/controllers/dynakube/extension/tls/conditions.go:3]: nitpick:
```suggestion
const conditionType string = "ExtensionsTLSSecret"
```

you are in the `tls` package, within the extensions folder, you don't really need to put the whole thing in the name :D

**Review by 0sewa0 — APPROVED** [review]

**Review by waodim — APPROVED** [review]
works for me, lgtm

File: pkg/controllers/dynakube/extension/tls/conditions.go (hunk 2 above)
Comment (albertogdd) [reply pkg/controllers/dynakube/extension/tls/conditions.go]: [Done](https://github.com/Dynatrace/dynatrace-operator/pull/3994/commits/b6ab3afa45839ba01fce54125c2c97c50025f602)

**Review by 0sewa0 — APPROVED** [review]

**Review by waodim — APPROVED** [review]

**Pull request merged**